### ISSUE TITLE
Incorrect port appended to hostname in base_uri of response body

### ISSUE THREAD

**Issue by PHameete** [confluentinc/kafka-rest#362]
Hi,

I am using the Confluent Kafka REST image. Because the default port is not available I've changed the port for the Kafka REST Proxy by changing the listeners configuration by setting the environment variable for the Kafka REST docker image:

`KAFKA_REST_LISTENERS: 'http://0.0.0.3503'`

However, when I create a consumer instance via the Kafka REST Proxy as follows:

```
curl -X POST -H "Content-Type: application/vnd.kafka.v2+json" -H "Accept: application/vnd.kafka.v2+json" \
    --data '{"name": "my_consumer_instance", "format": "json", "auto.offset.reset": "earliest"}' \
    http://schema_registry:3503/consumers/my_json_consumer
```

I get the following response:

```
{
    "instance_id": "my_consumer_instance",
    "base_uri": "http://schema_registry:8082/consumers/my_json_consumer/instances/my_consumer_instance"
}
```

Why is the Kafka REST Proxy not returning a base_uri with the port that it should be listening to? Am I missing some configuration (I've looked in the documentation but could not find any property for this) or is this indeed incorrect behavior?

Cheers!

**Comment by jaakkoo** [comment]
I'm facing the almost same issue. I'm running kafka-rest in mesos -cluster and the ports are assigned randomly  (well, from a pool of "what is free"). I'm running (or at least planning to) 3 instances of kafka-rest spread accross the cluster. 

I've set `host.name` to point to the the exact instance, but the port is never appended to `base_uri` in response, otherwise it seems fine.

If I only use an single instance everything seem to work fine. The reason why it works with a single instance is that i can always just talk to the lb and just throw the base_uri to /dev/null.

**Comment by mageshn** [comment]
@PHameete  are you testing using docker? That looks to be a bug to me. Can you try using the host.name and port config instead of the listener and let me know if it works

**Comment by PHameete** [comment]
@mageshn I am testing using Docker indeed. Im unable to test right now but if I remember correctly using host.name will result in a deprecated warning telling me to use the listener configuration and the container not starting.

**Comment by fbascheper** [comment]
This is still an issue with CP 4.0.0. 
The greatest problem is that the port cannot be set either using 
````
      -e KAFKA_REST_PORT=32380 
````
because that will produce the error "PORT is deprecated. Please use KAFKA_REST_LISTENERS instead." Basically this means that the port of this image is fixed to 8082. Anything else just will not work. 

Looks like https://github.com/confluentinc/cp-docker-images/issues/286 is a duplicate of this issue, BTW. d

**Comment by mageshn** [comment]
@fbascheper and @PHameete  This is getting fixed with https://github.com/confluentinc/kafka-rest/pull/374 . In the interim, I don't see an issue with using KAFKA_REST_PORT. Rest Proxy would throw a warning message but that should be fine. This would be the workaround until the bug fix is released.

**Comment by fbascheper** [comment]
Yes that would have been the work around... but it doesn't work because of the exit(1) straight after the warning message that port is deprecated. The Docker container immediately terminates because of the exit(1)

**Comment by fbascheper** [comment]
Mind that my comments relate to the Docker images of the Kafka rest. The exit command can be found in the following file: https://github.com/confluentinc/cp-docker-images/blob/4.0.x/debian/kafka-rest/include/etc/confluent/docker/configure

**Comment by imriz** [comment]
Is this bug ever got fixed? I still see this issue with latest docker images. advertised.listeners doesn't works, and I cannot set the rest port due to the issue mentioned in [this comment](https://github.com/confluentinc/kafka-rest/issues/362#issuecomment-360573887)
